### PR TITLE
Feature: setup java api

### PR DESCRIPTION
**Pull request by Zorin95670**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,105 +1,82 @@
...
-### lint
+```shell
+docker run -p 5432:5432 --rm -ti --name postgres -e POSTGRES_USER=leto_admin -e POSTGRES_PASSWORD=password -e POSTGRES_DB=leto_db postgres:16
+```
 
-Run eslint to check on the project.
+Once database is setup and the dev.env file is filled (cf: README.md).
 
-### lint:fix
+you can run:
```
Comment (justinehell) [inline CONTRIBUTING.md:39]: One sentence like :
Once database is setup and the dev.env file is filled (cf: README.md), you can run :

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,105 +1,82 @@
...
 
-Run eslint to fix on the project.
+```shell
+./gradlew bootrun
+```
 
-### lint:report
+> :warning: **You have to generate keystore.jks before!**
```
Comment (justinehell) [inline CONTRIBUTING.md:45]: maybe add link to the corresponding paragraph in README.md ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,105 +1,82 @@
...
 
-Generate issues report with eslint for sonar.
+## Checkstyle
 
-### test
+Before, push you branch and open/synchronize a pull-request. You have to verify the checkstyle of your application, here is the command:
```
Comment (justinehell) [inline CONTRIBUTING.md:49]: Before pushing your branch and open/synchronize a pull-request, you have to verify the checkstyle of your application. Here is the command to do so:

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,105 +1,82 @@
...
 
-Run all the unit tests.
+```shell
+./gradlew checkstyleMain
+```
 
-### test:coverage
+Reports can be find in:
```
Comment (justinehell) [inline CONTRIBUTING.md:55]: Reports can be found in:

File: README.md [hunk 5]
```diff
@@ -88,119 +108,52 @@ docker-compose down
...
+| DATABASE_PORT           | No, default: `5432`                    | A configuration parameter that defines the port number on which the database server is listening for connections.                   |
+| DATABASE_NAME           | No, default: `leto_db`                 | A configuration parameter that specifies the name of the specific database to be accessed on the server.                            |
+| DATABASE_USER           | No, default: `leto_admin`              | A configuration parameter indicating the username used to authenticate with the database server.                                    |
+| DATABASE_PASSWORD       | No, default: `password`                | A configuration parameter that specifies the password for authenticating the designated user with the database server.              |
+| SSL_KEY_STORE           | No, default: `classpath:keystore.jks`  | a configuration parameter specifying the file path to the keystore containing SSL/TLS certificates and keys for secure connections. |
```
Comment (justinehell) [inline README.md:122]: a configuration -> A configuration...

File: README.md [hunk 6]
```diff
@@ -88,119 +108,52 @@ docker-compose down
...
+DATABASE_PASSWORD=password
+SSL_KEY_STORE=classpath:keystore.jks
+SSL_KEY_STORE_PASSWORD=password
+SSL_KEY_PASSWORD=password
 ```
 
 See Configuration section for more details.
 
-Then build your docker image.
+Then run your docker compose.
```
Comment (justinehell) [inline README.md:151]: Then run your docker compose with this command:

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+# Actuator configuration
+management.endpoints.web.exposure.include=health
+
+# Server configuration
+## Https configuration
+server.port=8443
+server.servlet.context-path=/api
+server.ssl.key-store=${SSL_KEY_STORE:classpath:keystore.jks}
+server.ssl.key-store-password=${SSL_KEY_STORE_PASSWORD:password}
+server.ssl.key-password=${SSL_KEY_PASSWORD:password}
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/main/resources/application.properties:21]: done

File: CONTRIBUTING.md (hunk 3 above)
Comment (Zorin95670) [reply CONTRIBUTING.md:49]: done

File: CONTRIBUTING.md (hunk 4 above)
Comment (Zorin95670) [reply CONTRIBUTING.md:55]: done

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,105 +1,82 @@
...
+- `build/reports/checkstyle/main.xml`
 
-Run all the unit tests and generate coverage report of the unit tests for sonar.
+## Dependencies update
 
-### test:e2e
+You can check dependencies of your code, by running this command:
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply CONTRIBUTING.md:60]: done

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,105 +1,82 @@
...
+./gradlew dependencyUpdates
+```
 
-## Development
 
-### Directory structure
+## Generate test coverage report
 
-This is the default directory structure we use for the project:
+You can check dependencies of your code, by running this command:
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply CONTRIBUTING.md:69]: done

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,105 +1,82 @@
...
-│ ├ i18n               ⇨ Contains translation engine and translation files
-│ ├ layouts            ⇨ Contains all the page layouts
-│ ├ pages              ⇨ Contains all the pages
-│ ├ router             ⇨ Contains the router engine and the route definitions
-│ └ stores             ⇨ Contains store engine and all storage definitions
-└ tests                ⇨ Contains all files related to the tests
-```
 
-### How to release
+Reports can be find in:
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply CONTRIBUTING.md:75]: done

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -112,22 +89,22 @@ Steps to release:
 - Once the pull request validated and merged, tag the `main` branch with `vX.Y.Z`.
 - After the tag is pushed, make the release on the tag in GitHub.
 
-### Git: Default branch
+## Git: Default branch
 
 The default branch is main. Direct commit on it is forbidden. The only way to update the application is through pull request.
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply CONTRIBUTING.md:94]: done

File: README.md [hunk 12]
```diff
@@ -1,41 +1,54 @@
...
-- `CF_createDiagram`: allows user to create a diagram in `leto-modelizer`.
-- `CF_createDiagramFromTemplate`: allows user to create a diagram from template in `leto-modelizer`.
-- `CF_deleteDiagram`: allows user to delete a diagram in `leto-modelizer`.
-- `CF_createComponent`: allows user to create a component in `leto-modelizer`.
-- `CF_createComponentFromTemplate`: allows user to create a component from template in `leto-modelizer`.
-- `CF_deleteComponent`: allows user to delete a component in `leto-modelizer`.
-- `CF_useTextEditor`: allows user to use the text editor in `leto-modelizer`.
+List of permissions:
+
+- `admin`: allow user to access and use `leto-modelizer-admin`.
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply README.md:30]: done

File: README.md (hunk 6 above)
Comment (Zorin95670) [reply README.md:151]: done

File: CONTRIBUTING.md (hunk 2 above)
Comment (Zorin95670) [reply CONTRIBUTING.md]: done

**Review by justinehell — APPROVED** [review]

**Pull request merged**